### PR TITLE
Building web server with vlang

### PR DESCRIPTION
**Pull request by Umoren**
### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by Umoren** [comment]
The fixes https://github.com/section-engineering-education/engineering-education/pull/1771/commits/2704dfa3d35175d1720c31c20a78c6f2808a3a00

**Comment by geoffrey45** [comment]
@Umoren  Thank you for submitting your first article. 

For the author details, please create a new folder at the `authors` folder with the following files:
1. `index.md` -- this file carries your bio
2. `avatar.jpg` -- this is your avatar

Feel free to look at other students' folders to get the overview. You can also duplicate one of the author folders and edit it accordingly.

**Review by geoffrey45 — COMMENTED** [review]
> The fixes [2704dfa](https://github.com/section-engineering-education/engineering-education/commit/2704dfa3d35175d1720c31c20a78c6f2808a3a00)

Please add this line to the pull request header (the part with the checkboxes).  To link an issue, this should look like this: `This fixes #...` (note the #)

File: articles/building-your-web-server-with-vlang/index.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+Next, you start the webserver and listen to a port using`fn run<T>(port int)` . 
+
+Finally, to create routes in V, you can either specify the path as an attribute like this `["/"]` or auto-mapping of function names like this `index()` .
+
+ Compile the program using the v compiler, run `v run server.v` on your terminal.
+
+You should get something like this on your terminal:
+
+![Building%20your%20Web%20Server%20with%20Vlang%2055ee48a335b145df9af2fe6d5f312905/Untitled.png](/articles/building-your-web-server-with-vlang/v-server.png)
+
```
Comment (geoffrey45) [inline articles/building-your-web-server-with-vlang/index.md:236]: Please format your image to look like this: 
```markdown
![vlang server](/engineering-education/building-your-web-server-with-vlang/v-server.png)
```
Please format all your images as shown above.

**Review by geoffrey45 — CHANGES_REQUESTED** [review]

File: articles/building-your-web-server-with-vlang/index.md [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+Let's go through some features of V.
+
+#### Code Style
+
+The V programming language drives programmers to write good code. There’s only one programming style with V unlike languages with similar syntax such as Rust and Golang that have functional and object-oriented styles of writing programs. V claims that cause of its one-style approach to programming, it’s easier for developers in teams to understand, change team members’ code, and build maintainable software.
+
+#### Safety
+
+V is very strict when it comes to writing safe programs. Some language policies are:
+
```
[the remark this answers is not in the document]
Comment (Umoren) [reply articles/building-your-web-server-with-vlang/index.md:44]: Okay, so there's no formal definition for the term "safe program". I was trying to explain why i think V is safe in the language policies. 
Anyway, i rewrote the sentences: `V is very strict with writing secure, fast and memory-safe programs. Some strict policies of the language are`

Is this okay?

**Comment by Umoren** [comment]
This fixes #1771

**Comment by Umoren** [comment]
@geoffrey45 I've made the requested changes

**Comment by Umoren** [comment]
@hectorkambow i've made the requested changes

File: articles/building-your-web-server-with-vlang/index.md [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+	mut:
+		count int
+}
+```
+
+and then make use of `count` somewhere else in the program.
+
+Next, you start the webserver and listen to a port using`fn run<T>(port int)` . 
+
+Finally, to create routes in V, you can either specify the path as an attribute like this `["/"]` or auto-mapping of function names like this `index()` .
```
Comment (geoffrey45) [inline articles/building-your-web-server-with-vlang/index.md:229]: Explain what `fn run<T>(port int)` is. Provide examples showing creating different routes using the methods above.

**Review by geoffrey45 — CHANGES_REQUESTED** [review]
@Umoren Please make the changes below and ping me when you are done. 

We tend to avoid using product logos as hero images. Please replace your `v-logo.png` with a 600 x 400 image from [Unsplash.com](https://unsplash.com), named `hero.png`  or `hero.jpg`. You can choose the download size when downloading. Feel free to create a custom hero image. 

Your images are not legible. Please replace your images with HD images. If not, you can omit them entirely and replace them with code output.

**Comment by Umoren** [comment]
The fixes #https://github.com/section-engineering-education/engineering-education/pull/1771

**Comment by Umoren** [comment]
@geoffrey45 
I've made the requested changes.

File: articles/building-your-web-server-with-vlang/index.md [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+```c
+fn (r Repo) find_user_by_id(id int) ?User {
+	for user in r.users {
+		if user.id == id {
+			// V automatically wraps this into an option type
+			return user
+		}
+	}
+	return error('User $id not found')
+}
```
[the remark this answers is not in the document]
Comment (geoffrey45) [reply articles/building-your-web-server-with-vlang/index.md:105]: @Umoren This one

**Comment by geoffrey45** [comment]
@Umoren Your article's image shoub be named `hero.jpg`. Please rename the image as such.

**Comment by Umoren** [comment]
Oh this doesn't have an output. It's a helper function.
The code snippet below shows the output

```c++
fn main() {
	repo := Repo{
		users: [User{1, 'Andrew'}, User{2, 'Bob'}, User{10, 'Charles'}]
	}
	user := repo.find_user_by_id(10) or { // Option types must be handled by `or` blocks
		return
	}

  user := repo.find_user_by_id(7) or {
    println(err) // "User 7 not found"
    return
  }

	println(user.id) // "10"
	println(user.name) // "Charles"
}

```
`

**Comment by Umoren** [comment]
@geoffrey45 I've renamed the article image.
Thanks

**Comment by hectorkambow** [comment]
plagiarism check done

**Review by geoffrey45 — APPROVED** [review]

**Comment by Umoren** [comment]
Hi @geoffrey45 
To avert issues with payment, I have changed my Paypal link to this: 
[https://paypal.me/DanielWildegger](https://paypal.me/DanielWildegger)
Thanks.

**Comment by geoffrey45** [comment]
> Hi @geoffrey45 
> To avert issues with payment, I have changed my Paypal link to this: 
> [https://paypal.me/DanielWildegger](https://paypal.me/DanielWildegger)
> Thanks.

@hectorkambow

**Pull request merged**